src/index.d.ts

```diff
@@ -37,9 +37,9 @@ export interface VNode<P = {}> {
 
 export type Key = string | number | any;
 
-export type RefObject<T> = { current: T | null };
+export type RefObject<T> = { current: T };
 export type RefCallback<T> = (instance: T | null) => void;
-export type Ref<T> = RefObject<T> | RefCallback<T> | null;
+export type Ref<T> = RefCallback<T> | RefObject<T | null> | null;
```

Comment:
[React for reference](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9efe05446271637b014cc445c892923a4f4510ef/types/react/index.d.ts#L192)